arch/arm/src/stm32f7/stm32_adc.c

```diff
@@ -1137,20 +1394,22 @@ static void adc_rccreset(struct stm32_dev_s *priv, bool reset)
 
 static void adc_enable(struct stm32_dev_s *priv, bool enable)
 {
+  bool enabled = false;
+
   ainfo("enable: %d\n", enable ? 1 : 0);
 
-  if (enable)
+  if (!enabled && enable)
     {
       adc_modifyreg(priv, STM32_ADC_CR2_OFFSET, 0, ADC_CR2_ADON);
     }
-  else
+  else if (enabled && !enable)
```

Comment:
what is the point here? `else if (enabled && !enable)` is always evaluated to `false`. Why that is needed?

Reply:
Remains from arch/stm32 which I missed. Removed